docs/dev/Pagination-v2.md

```diff
@@ -285,3 +285,87 @@ OpenSearchExecutionEngine->>+ProjectOperator: getTotalHits
     ResourceMonitorPlan-->>-ProjectOperator: value
   ProjectOperator-->>-OpenSearchExecutionEngine: value
 ```
+
+### Plan Tree changes
+
+#### Abstract Plan tree
+
+Changes:
+1. New Plan node -- `Paginate` -- added into the tree.
+2. `QueryPlan` got new optional field: page size. When it is set, `Paginate` is being added. It is converted to `LogicalPaginate` later. For non-paging requests the tree remains unchanged.
```

Comment:
int is not an Optional.  Please consider making `pageSize` an Optional<Integer> instead of including an 'invalid' page size int.